### PR TITLE
Doubles: Fix Aromatherapy/Sap Sipper interaction 

### PR DESCRIPTION
**Pull request by pacmanboss256**
aromatherapy now boosts allies with sap sipper in double battles

### CODE REVIEW

File: data/moves.ts [hunk 1]
```diff
@@ -504,8 +504,11 @@ export const Moves: {[moveid: string]: MoveData} = {
 			this.add('-activate', source, 'move: Aromatherapy');
 			let success = false;
 			for (const ally of pokemon.side.pokemon) {
-				if (ally !== source && ((ally.hasAbility('sapsipper')) ||
-						(ally.volatiles['substitute'] && !move.infiltrates))) {
+				if (ally !== source && (ally.hasAbility('sapsipper'))) {
```
Comment (AnnikaCodes) [inline data/moves.ts:507]:
```suggestion
				if (ally !== source && ally.hasAbility('sapsipper')) {
```

File: data/moves.ts [hunk 2]
```diff
@@ -504,8 +504,11 @@ export const Moves: {[moveid: string]: MoveData} = {
...
+					ally.boostBy({atk: 1});
+					continue;
+				}
+				if (ally !== source && ((ally.volatiles['substitute'] && !move.infiltrates))) {
```
Comment (AnnikaCodes) [inline data/moves.ts:511]:
```suggestion
				if (ally !== source && ally.volatiles['substitute'] && !move.infiltrates) {
```
`&&` is associative

**Comment by urkerab** [comment]
> aromatherapy now boosts allies with sap sipper in double battles

When you say "now", in which game did the behaviour change?

**Comment by pacmanboss256** [comment]
should be gen 8

**Comment by Marty-D** [comment]
Wasn't the bug reported that this interaction only works sometimes? This change seems like it'll make it work all the time and sometimes twice.

**Comment by pacmanboss256** [comment]
The bug report just says it doesn't work, nothing about partially working

**Comment by DaWoblefet** [comment]
There is a skipped test for this behavior in tests/sim/abilities/sapsipper.js. Make sure to unskip to check if it passes that.

I am pretty sure the bug is slot-dependent.

**Comment by pacmanboss256** [comment]
i ran the test and it passed

**Comment by Marty-D** [comment]
So, yes
```
Turn 1

Goodra used Sleep Talk!
But it failed!

The opposing Goodra used Sleep Talk!
But it failed!

Blissey used Aromatherapy!
[Goodra's Sap Sipper]
Goodra's Attack rose!
A soothing aroma wafted through the area!
But it failed!
[DEBUG] move failed because it did nothing.

The opposing Blissey used Aromatherapy!
A soothing aroma wafted through the area!
But it failed!
[DEBUG] move failed because it did nothing.
```
The first Blissey that moved was to the right of its ally Goodra, which is why it worked.

**Pull request closed without merging**